src/envoy/utils/mixer_control.h

```diff
@@ -42,5 +48,30 @@ Grpc::AsyncClientFactoryPtr GrpcClientFactoryForCluster(
     const std::string &cluster_name, Upstream::ClusterManager &cm,
     Stats::Scope &scope);
 
+bool ExtractNodeInfo(const envoy::api::v2::core::Node &node, LocalNode *args);
+
+inline bool ReadProtoMap(
+    const google::protobuf::Map<std::string, google::protobuf::Value> &meta,
+    const std::string &key, std::string *val) {
+  const auto it = meta.find(key);
+  if (it != meta.end()) {
+    *val = it->second.string_value();
+    return true;
+  }
+
+  return false;
+}
+
+inline bool ReadAttributeMap(
```

Comment:
Are these two "read" functions used in other files?   If no,  prefer to not in the header file.  It is polluting .